### PR TITLE
feat(fhir): add US Core meta.profile to exported resources

### LINKED ISSUE #938: fhir: add meta.profile (US Core) to exported resources
## Context

PR #922 (https://github.com/blamechris/carebridge/pull/922) added the `Encounter` and `Procedure` generators, but — like the other resource generators — it does not populate `resource.meta.profile` with a US Core profile URL.

If the FHIR gateway ever claims US Core conformance (which most US healthcare exchange partners will expect), each exported resource needs to assert the specific US Core profile it conforms to:

- `Encounter` → `http://hl7.org/fhir/us/core/StructureDefinition/us-core-encounter`
- `Procedure` → `http://hl7.org/fhir/us/core/StructureDefinition/us-core-procedure`
- `Patient`, `Condition`, `MedicationStatement` (+ `MedicationRequest`), `AllergyIntolerance`, `Observation` — each with its corresponding US Core URL

## Scope

Cross-cutting — affects every generator under `services/fhir-gateway/src/generators/`. Likely pairs with validating the resources against the relevant US Core profile (e.g. via HL7 validator in CI).

## Acceptance

- [ ] Decide whether CareBridge's FHIR export is claiming US Core conformance (product/compliance call)
- [ ] If yes: populate `meta.profile` on every exported resource
- [ ] Add validator-based test coverage (or at minimum unit tests asserting the profile URL is present)

Discovered during review of PR #922.

### PR DESCRIPTION
**Pull request by blamechris**
## Summary
- Populate `resource.meta.profile` on every exported FHIR resource with the appropriate US Core StructureDefinition URL.
- Centralise URL constants in `services/fhir-gateway/src/generators/us-core-profiles.ts` so every generator references the same canonical strings.
- Practitioner is excluded — covered by #947.
- MedicationStatement is excluded — US Core 5+ deprecated the profile; emitting any URL would be a false conformance claim.

## Profile choices
| Resource | Profile URL |
| --- | --- |
| Patient | `http://hl7.org/fhir/us/core/StructureDefinition/us-core-patient` |
| Condition | `http://hl7.org/fhir/us/core/StructureDefinition/us-core-condition-encounter-diagnosis` |
| MedicationRequest | `http://hl7.org/fhir/us/core/StructureDefinition/us-core-medicationrequest` |
| MedicationStatement | *(omitted — no current US Core profile)* |
| AllergyIntolerance | `http://hl7.org/fhir/us/core/StructureDefinition/us-core-allergyintolerance` |
| Observation (vital) | `http://hl7.org/fhir/us/core/StructureDefinition/us-core-vital-signs` |
| Observation (lab) | `http://hl7.org/fhir/us/core/StructureDefinition/us-core-laboratory-result-observation` |
| Encounter | `http://hl7.org/fhir/us/core/StructureDefinition/us-core-encounter` |
| Procedure | `http://hl7.org/fhir/us/core/StructureDefinition/us-core-procedure` |

### Notes on the Condition choice
The condition generator does not currently emit a `category` coding, so we cannot programmatically disambiguate `us-core-condition-encounter-diagnosis` vs `us-core-condition-problems-and-health-concerns`. The encounter-diagnosis profile is the more conservative default given that the internal `diagnoses` table is sourced from clinical encounters. If a future change adds a problem-list category, the condition generator can branch on it the same way the observation generator already does for vital-signs vs laboratory.

### Notes on the Observation branching
Vital observations and lab observations are produced by two separate exported functions (`toFhirVitalObservation` and `toFhirLabObservation`), so the branch on profile URL falls out naturally — no runtime category lookup required.

## Test plan
- [x] `pnpm --filter @carebridge/fhir-gateway test` — 27 files, 435 tests pass (was 26/424; +11 tests covering meta.profile on each generator including the deliberate MedicationStatement omission)
- [x] `pnpm typecheck`
- [x] `pnpm lint`

Closes #938.

### CODE REVIEW

**Comment by blamechris** [comment]
## Code Review: PR #1137

### Summary
Populates `resource.meta.profile` on every exported FHIR resource (Patient, Condition, MedicationRequest, AllergyIntolerance, Observation [vital + lab], Encounter, Procedure) with the appropriate US Core StructureDefinition URL. URLs are centralised in a new `services/fhir-gateway/src/generators/us-core-profiles.ts`. MedicationStatement deliberately omits `meta.profile` (US Core 5+ deprecated the profile); Practitioner is out of scope and tracked under #947 / #1141. +382 / -5, +11 tests (424 → 435), all 3 CI checks green.

### Strengths
- **Single source of truth.** `us-core-profiles.ts` collects every URL behind a shared `BASE` constant — no chance of drift between generators, easy to audit the conformance surface.
- **Conservative profile selection.** The Condition choice (`us-core-condition-encounter-diagnosis`) is correctly justified in the PR body and inline comment: the `diagnoses` table is sourced from encounters and lacks a category coding to disambiguate. The follow-up path (branch on category once it exists) is also called out.
- **Observation per-function branching.** Splitting on the exported function (`toFhirVitalObservation` vs. `toFhirLabObservation`) rather than a runtime category lookup is the right call — no `if` on data, the function name is the type.
- **Honest MedicationStatement omission.** Test explicitly asserts `meta` is `undefined` rather than empty, which prevents a future drive-by from quietly adding a wrong profile URL.
- **Negative test.** `does NOT emit the vital-signs profile on a lab observation` catches the most likely regression in this area.
- **Base-URL guard.** `profile URL constants` test enforces every URL starts with the canonical US Core `StructureDefinition/` prefix — cheap and effective.

### Issues Found

#### Critical (Must Fix)
None.

#### Suggestions (Should Consider)
None within this PR's scope — both raised items are deferred to follow-ups (see below).

#### Nitpicks (Optional)
- `us-core-profiles.ts` exports each constant as a backtick template literal (`` `${BASE}/us-core-patient` ``) — they're all compile-time constants, so plain string literals would read marginally cleaner. Not worth churning.
- `makeMedication()` in the test file uses both `2026-04-10` and the test was written assuming current date; intentional given the project's `currentDate` is 2026, but worth noting for anyone diffing this file in isolation.

### Profile URL audit (vs. HL7 US Core IG)
| Resource | URL | Verdict |
| --- | --- | --- |
| Patient | `…/us-core-patient` | Correct |
| Condition | `…/us-core-condition-encounter-diagnosis` | Correct (US Core 5+ split; encounter-diagnosis is the right conservative default given no category) |
| MedicationRequest | `…/us-core-medicationrequest` | Correct |
| MedicationStatement | *(omitted)* | Correct — US Core 5+ deprecated the profile; emitting any URL would be a false claim |
| AllergyIntolerance | `…/us-core-allergyintolerance` | Correct |
| Observation (vital) | `…/us-core-vital-signs` | Correct as umbrella; see #1145 for per-vital specialisation |
| Observation (lab) | `…/us-core-laboratory-result-observation` | Correct |
| Encounter | `…/us-core-encounter` | Correct |
| Procedure | `…/us-core-procedure` | Correct |

### Deferred Items (Follow-Up Issues)

| Suggestion | Issue | Rationale for deferral |
| --- | --- | --- |
| Emit specialised US Core vital-signs profiles per vital type (`us-core-blood-pressure`, `us-core-heart-rate`, `us-core-pulse-oximetry`, etc.) on top of the umbrella `us-core-vital-signs`. | [#1145](https://github.com/blamechris/carebridge/issues/1145) | Umbrella profile is fully conformant. Per-type slicing is a strict improvement but adds a non-trivial mapping table and test matrix — better as its own PR. |
| Promote locally-declared `FhirCondition` / `FhirAllergyIntolerance` / `FhirObservation` interfaces from their generator modules to `types/fhir-r4.ts`, matching the pattern already used by `FhirPatient` / `FhirEncounter` / `FhirProcedure` / `FhirMedicationRequest`. | [#1146](https://github.com/blamechris/carebridge/issues/1146) | Pre-existing inconsistency. Touching three more files in this PR would obscure the meta.profile change; better tracked as a focused refactor. |

### Architecture Notes
Lives entirely within `services/fhir-gateway`. No tRPC, no BullMQ, no DB schema impact. The Meta type already existed in `types/fhir-r4.ts` and is reused. ESM `.js` import discipline preserved. Functional style — no classes introduced.

### Verdict
- [x] **Approve** — Ready to merge

File: services/fhir-gateway/src/__tests__/us-core-meta-profile.test.ts [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+    notes: null,
+    created_at: "2026-04-10T10:00:00.000Z",
+  } as EncounterRow;
+}
+
+type ProcedureRow = Parameters<typeof toFhirProcedure>[0];
+function makeProcedure(): ProcedureRow {
+  return {
+    id: "p1",
+    patient_id: "pat1",
```
[the remark this answers is not in the document]
Comment (blamechris) [reply services/fhir-gateway/src/__tests__/us-core-meta-profile.test.ts]: **FIX**

Fixed in `4414f47`

**Change:** Aligned `makeProcedure()`'s `patient_id` with the `"p1"` passed to `toFhirProcedure(...)` so the fixture is internally consistent with every other fixture in this file (`makeDiagnosis`, `makeMedication`, `makeAllergy`, `makeVital`, `makeLab`, `makeEncounter` all use `"p1"`). This makes the fixture representative and avoids masking a future regression if the generator starts cross-checking `procedure.patient_id`.

```diff
 function makeProcedure(): ProcedureRow {
   return {
     id: "p1",
-    patient_id: "pat1",
+    patient_id: "p1",
     name: "Appendectomy",
```

Test still green: 11/11 in `us-core-meta-profile.test.ts`.

**Comment by blamechris** [comment]
## Review Comments Addressed

| # | Comment | Outcome | Reference |
|---|---------|---------|-----------|
| 1 | `makeProcedure()` fixture's `patient_id: "pat1"` mismatches the `"p1"` passed to `toFhirProcedure(...)` — make the fixture representative. | FIX | `4414f47` |

**Total:** 1 comment addressed
- Fixed: 1 (commit `4414f47`)
- False positives: 0
- Follow-up issues created: 0 (the two #1145 / #1146 follow-ups were created during the earlier agent-review pass, not from this check-pr round)
- Existing issues closed: 0

CI was green pre-fix (3/3) and the touched test file passes locally (11/11 in `us-core-meta-profile.test.ts`). Awaiting fresh CI run.
